proposed/2021/ShowVulnerabilitiesInThePMUI.md

```diff
@@ -0,0 +1,90 @@
+# Show NuGet Package Vulnerabilities in the Visual Studio Package Manager UI
+
+* Status: **MVP Implementation**
+* Author(s): [Chris Gill](https://github.com/chgill-MSFT)
+* Issue: 
+* Type: Feature
+
+## Description: What is it?
+
+When a vulnerability in a NuGet package is discovered, surface an indicator of the vulnerability in the Package Manager UI to alert developers and help them take the appropriate action.
+
+## Problem: What problem is this solving?
+
+Vulnerabilities in packages can be leveraged by malicious actors to do harm to developers and their users. At the time of writing this, there are 113 NuGet advisories in the [GitHub Advisory Database](https://github.com/advisories?query=ecosystem%3Anuget). While package vulnerabilities can be found on NuGet.org and with the dotnet list command, most developer interactions with NuGet occur in the Visual Studio package manager UI where no vulnerability information is currently available.
+
+## Why: How do we know this is a real problem worth solving?
+
+NuGet hosts over 200,000 packages that developers use today. Most packages depend on another package, to which the attack vector / blast radius of any security vulnerability can be catastrophic in any package management ecosystem.
+
+## Success: How do we know if we've solved this problem?
+
+We will use the following to measure the success of this feature:
+
+* We see a significant decrease in direct installations for vulnerable packages versions after surfacing vulnerability indicators in the PMUI.
+* We see a significant increase in updates and/or uninstalls for directly installed vulnerable package versions after surfacing vulnerability indicators in the PMUI.
```

Comment:
See this comment on @jebriede's technical spec: https://github.com/NuGet/Home/pull/11033#discussion_r670731753 -- it's worth noting that only when clicking on a package in the Browse tab will vulnerabilities (and deprecations) become apparent to the user (this is due to the search metadata lacking this info, as you call out below, and the need to load registration metadata to access it), but this is still a necessary step before installation, so it should work. Bullet 2 here, the installed pane, is populated with registration metadata included, so we will even see indicators on the Installed tab (without clicking on packages)

Reply:
Today the Browse tab also looks up deprecation/vulnerability from the registration and surfaces it on the package list items. You don't have to click on  a package to see that it's deprecated/vulnerable, only to see the details.

![image](https://user-images.githubusercontent.com/15097183/127570095-20e1530b-2f22-4996-957e-df5a50fe7a75.png)